.gitignore

```diff
@@ -256,3 +256,6 @@ doc-nits
 # LSP (Language Server Protocol) support
 .cache/
 compile_commands.json
+**/.DS_Store
+.vscode/settings.json
+.vscode/tasks.json
```

Comment:
Please do not add unrelated changes to a commit. If you want them here in the PR (although it would better be a separate PR), please at least separate them to a different commit.